juju/juju
# [JUJU-1091] Cloud-init wait for IP

The linked LP Bug appears to show that it is possible during cloud-init, to run `netplan apply` but not have a DHCP lease before attempting to proceed to apt updates, which can then fail.

Here we add some bash to cloud-init that will cause us to wait up to 5 seconds to be assigned an IP address.

## QA steps

- Add a LXD container to an existing machine.
- `juju ssh` to the host machine, then run `lxd exec <container-name> bash`
- Check in /var/log/cloud-init-output.log for a line like:
```

```

## Documentation changes

None

## Bug reference

https://bugs.launchpad.net/juju/+bug/1944080
